packages/back-end/src/models/SegmentModel.ts

```diff
@@ -15,9 +16,73 @@ const segmentSchema = new mongoose.Schema({
   dateCreated: Date,
   dateUpdated: Date,
 });
-export type SegmentDocument = mongoose.Document & SegmentInterface;
 
-export const SegmentModel = mongoose.model<SegmentDocument>(
-  "Segment",
-  segmentSchema
-);
+type SegmentDocument = mongoose.Document & SegmentInterface;
+
+const SegmentModel = mongoose.model<SegmentDocument>("Segment", segmentSchema);
+
+function toInterface(doc: SegmentDocument): SegmentInterface {
+  return doc.toJSON();
+}
+
+export async function createSegment(segment: Partial<SegmentInterface>) {
+  return toInterface(await SegmentModel.create(segment));
+}
+
+export async function findSegmentById(id: string, organization: string) {
+  // If using config.yml, immediately return the list from there
+  if (usingFileConfig()) {
+    return getConfigSegments(organization).filter((s) => s.id === id)[0];
+  }
```

Comment:
@jdorn Hopping back to this and have some questions...

So, the changes needed are, build an env variable `STORE_SEGMENTS_IN_MONGO`, and update the `SegmentModel` methods so we only get segments from the config file if `usingConfigFile()` is true and `process.env.STORE_SEGMENTS_IN_MONGO` is false. (This is simplified, and will likely require more changes than this, but this is just the overall idea)

Otherwise, we'll get the segments from Mongo.

We'll then need to figure out a way to alert orgs who have segments stored in Mongo, but are `usingConfigFile()` and don't have `process.env.STORE_SEGMENTS_IN_MONGO` set to true. In this case, they won't see any of the segments in mongo so we'll need to let them know they need to set the env variable if they want to use those segments.

Should we also give them a way to easily delete those segments in order to clear the warning? Or should we handle that on a case by case basis since the number of orgs affected might be super small.

Should we direct them to `/settings` and prompt them to export a new config.yml - in my environment, that will generate a new config.yml that contains those segments saved in Mongo - is that expected or just based on my environment?

Question 1: This should only affect self-hosted organizations, correct?

Question 2: When we release this feature, these self-hosted organizations don't get the updates automatically, correct? They need to manually increment the version of GB they're using, right? Or is there a possibility that some of these self-hosted orgs have something in place to automatically upgrade to the latest? I mainly want to know if we're going to potentially introduce a breaking change for the affect self-hosted orgs. Will someone be attempting to use GrowthBook and get blocked until their dev team adds the env variable?

Reply:
Sending them to `/settings` won't help if they are currently using `config.yml`.  We actually disable exporting entirely in that case since it's not very useful.  All of the Model methods read from the config file, so exporting wouldn't include anything in the database anyway.

As for letting them dismiss the error, that shouldn't be necessary.  We should only show the error if there are no segments.  Once they add at least one segment to config.yml, the error message should stop showing up.

We can actually make the logic a lot simpler than you are describing.  No need to query mongo to check for records at all.  Just show the error message telling them to add the env variable if the following is true:

```ts
const showConfigError = 
  usingConfigFile() 
  && !process.env.STORE_SEGMENTS_IN_MONGO 
  && !segments.length
```

For you questions:

1. Yes, this is only for self-hosted.  `usingConfigFile()` will always return false on Cloud, so you shouldn't need anything special in your code to handle this.
2. Some users have auto-update enabled, but most likely update manually.  There is a chance this could block a user, but I don't think it's a huge deal.  The number of people using both config.yml and segments today is pretty small and segments are more of an optional feature.  You can still do 99% of things in GrowthBook without having segments defined.

Reply:
@jdorn Just pushed up changes on the front-end to do this check in `segments/index.tsx` along with some doc changes. I haven't been able to test this end-to-end (struggling to get my env set up so that I can get myself into this state). 

Is this what you were thinking?